backend/lm_backend/api/routes/jobs.py

```diff
@@ -23,11 +27,28 @@
     dependencies=[Depends(guard.lockdown(Permissions.JOB_EDIT))],
 )
 async def create_job(
-    job: JobCreateSchema,
+    job: JobWithBookingCreateSchema,
     db_session: AsyncSession = Depends(get_session),
 ):
     """Create a new job."""
-    return await crud_job.create(db_session=db_session, obj=job)
+    job_created: Job = await crud_job.create(
+        db_session=db_session, obj=JobCreateSchema(**job.dict(exclude={"bookings"}))
+    )
+
+    if job.bookings:
+        for booking in job.bookings:
+            obj = {
+                "job_id": job_created.id,
+                "feature_id": booking.feature_id,
+                "quantity": booking.quantity,
+            }
+            try:
+                await crud_booking.create(db_session=db_session, obj=BookingCreateSchema(**obj))
+            except HTTPException:
+                await crud_job.delete(db_session=db_session, id=job_created.id)
+                raise
```

Comment:
I think you could wrap the whole block (starting on line 39 with the try/except. Since you are going to blow it up if one fails, you don't need to handle the exception inside the loop.

Reply:
That makes sense, I updated it.